File: java/bank_simulation/src/main/java/com/nursyah/bank_simulation/features/admin/IAdminService.java
package com.nursyah.bank_simulation.features.admin;

import com.nursyah.bank_simulation.features.admin.dto.RequestTopUpCustomer;
import com.nursyah.bank_simulation.shared.dto.RequestPaginated;
import com.nursyah.bank_simulation.shared.dto.ResponseDto;
import com.nursyah.bank_simulation.shared.dto.ResponsePaginated;
import com.nursyah.bank_simulation.shared.dto.ResponseTransaction;
import com.nursyah.bank_simulation.shared.dto.ResponseUser;
import java.util.concurrent.CompletableFuture;

public interface IAdminService {
    CompletableFuture<ResponseDto> topUpCustomerBalanceAsync(RequestTopUpCustomer request);
    CompletableFuture<ResponsePaginated<ResponseUser>> viewCustomersAsync(RequestPaginated request);
    CompletableFuture<ResponsePaginated<ResponseTransaction>> viewTransactionCustomersAsync(RequestPaginated request);
}

File: dotnet/bank_simulation/Features/Admin/IAdminService.cs
using bank_simulation.Features.Admin.Dto;
using bank_simulation.Shared.Dto;

namespace bank_simulation.Features.Admin;

public interface IAdminService
{
    Task<ResponseDto> TopUpCustomerBalanceAsync(RequestTopUpCustomer request);
    Task<ResponsePaginated<ResponseUser>> ViewCustomersAsync(RequestPaginated request);
    Task<ResponsePaginated<ResponseTransaction>> ViewTransactionCustomersAsync(RequestPaginated request);
}